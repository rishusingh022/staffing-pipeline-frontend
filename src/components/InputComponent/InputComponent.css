.login-input{
    width: 289.71px;
    height: 36.48px;
    background: #D9D9D9;
    padding: 10px;
}