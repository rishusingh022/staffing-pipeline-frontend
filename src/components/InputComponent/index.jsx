import React from 'react';

import './InputComponent.css';

import propTypes from 'prop-types';

export default function InputComponent({ placeholder }) {
  return (
    <div>
      <input type="text" className={'login-input'} placeholder={placeholder} />
    </div>
  );
}

InputComponent.propTypes = {
  placeholder: propTypes.string,
};
