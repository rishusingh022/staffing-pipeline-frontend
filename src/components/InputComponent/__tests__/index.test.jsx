import React from 'react';

import { render, screen } from '@testing-library/react';

import InputComponent from '..';

describe('InputComponent', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<InputComponent placeholder="email" className="login-input" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display placeholder value passed to it', () => {
    render(<InputComponent placeholder="email" className="login-input" />);
    expect(screen.getByPlaceholderText('email')).toBeTruthy();
  });
});
