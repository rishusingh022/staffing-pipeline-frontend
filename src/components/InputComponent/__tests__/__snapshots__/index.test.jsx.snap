// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComponent should render correctly 1`] = `
<DocumentFragment>
  <div>
    <input
      class="login-input"
      placeholder="email"
      type="text"
    />
  </div>
</DocumentFragment>
`;
